### PR TITLE
GODRIVER-1530 GODRIVER-1213 Fix the enterprise auth test runner.

### PR DESCRIPTION
**Pull request by matthewdale**
[GODRIVER-1530](https://jira.mongodb.org/browse/GODRIVER-1530), [GODRIVER-1213](https://jira.mongodb.org/browse/GODRIVER-1213)

## Summary
* Refactor the enterprise auth test runner to use `mongo.Connect` instead of `topology.New`.
* Load the MongoDB test URI from the `MONGODB_URI` environment variable and ensure the URI environment variable is always passed to the test runner.
* Move the test runner to the `cmd/` directory. Remove all outdated examples from the `x/mongo/driver/examples` directory.
* Allow the `SERVICE_HOST` auth mechanism property to be specified in a connection string.

## Background & Motivation
https://github.com/mongodb/mongo-go-driver/pull/1044 broke the enterprise auth tests, which are only run on the waterfall and went unnoticed until now. That PR broke the tests because it didn't pull the database name from the connection string, instead it used "test" for all tests. In any case, we shouldn't use the unstable `topology.Topology` APIs for the enterprise auth tests anyway because it's not representative of how most people use enterprise auth (most use the `mongo.Client` type via `mongo.Connect`).

Refactor the enterprise auth test runner to use `mongo.Connect` instead of `topology.New`, loading the MongoDB test URI from the `MONGODB_URI` environment variable and using the `defaultauthdb` database name from the connection string, and move the test runner to the `cmd/` directory. Remove all outdated examples from the `x/mongo/driver/examples` directory.

### CODE REVIEW

File: cmd/testentauth/main.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
```
Comment (benjirewis) [inline cmd/testentauth/main.go:1]:
```suggestion
// Copyright (C) MongoDB, Inc. 2022-present.
```

**Review by benjirewis — CHANGES_REQUESTED** [review]
Evergreen changes and the new `testentauth/main.go` LGTM. 

Those examples definitely look outdated; I think I'm fine removing, as I don't see what to replace the examples with. But, can we also close GODRIVER-2527 and remove the `x/mongo/driver/examples` folder entirely?

**Review by prestonvasquez — APPROVED** [review]
LGTM

File: cmd/testentauth/main.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	coll := client.Database(cs.Database).Collection("test")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	count, err := coll.EstimatedDocumentCount(ctx)
+	if err != nil {
+		log.Fatalf("failed executing count command: %v", err)
+	}
+	log.Println("Count of test.test:", count)
```
Comment (benjirewis) [inline cmd/testentauth/main.go:50]:
```suggestion
	log.Println("Count of test collection:", count)
```
May not be `test.test`.

**Review by benjirewis — APPROVED** [review]
LGTM assuming tests pass, thank you.

**Pull request merged**